src/ch/imedias/rsccfx/model/Rscc.java

```diff
@@ -15,22 +24,79 @@
    * Important: Make sure to NOT include a / in the beginning or the end.
    */
   private static final String DOCKER_FOLDER_NAME = "docker-build_p2p";
-  private final String pathToResourceDocker;
-  private final String scriptShell;
-  private final ProcessExecutor processExecutor;
+  private static final String RSCC_FOLDER_NAME = ".rscc";
+  private SystemCommander systemCommander;
+  private String pathToResourceDocker;
   private StringProperty key = new SimpleStringProperty();
   private String keyServerIp;
   private String keyServerHttpPort;
 
   /**
    * Initializes the Rscc model class.
    */
-  public Rscc(ProcessExecutor processExecutor) {
-    this.processExecutor = processExecutor;
-    pathToResourceDocker =
-        getClass().getClassLoader().getResource(DOCKER_FOLDER_NAME)
-            .getFile().toString().replaceFirst("file:", "");
-    scriptShell = "bash" + " " + pathToResourceDocker + "/";
+  public Rscc(SystemCommander systemCommander) {
+    this.systemCommander = systemCommander;
+    defineResourcePath();
+  }
+
+  /**
+   * Sets resourcepath - dependet if application runs as JAR or in IDE.
+   */
+  private void defineResourcePath() {
+    String userhome = System.getProperty("user.home");
+    URL location = this.getClass().getProtectionDomain().getCodeSource().getLocation();
+    File f = new File(location.getFile());
```

Comment:
I can't find a matching name since I don't know what gets achieved with this variable